lucene/core/src/java/org/apache/lucene/search/TopFieldCollector.java

```diff
@@ -123,13 +124,22 @@ public LeafCollector getLeafCollector(LeafReaderContext context) throws IOExcept
         @Override
         public void setScorer(Scorable scorer) throws IOException {
           super.setScorer(scorer);
+          minCompetitiveScore = 0f;
           updateMinCompetitiveScore(scorer);
+          if (minScoreAcc != null) {
+            updateGlobalMinCompetitiveScore(scorer);
+          }
         }
 
         @Override
         public void collect(int doc) throws IOException {
           ++totalHits;
           hitsThresholdChecker.incrementHitCount();
+
+          if (minScoreAcc != null && totalHits % minScoreAcc.modInterval == 0) {
```

Comment:
yep no idea, I do it like that in order to be able to change the interval in tests but if it hurt performance we might want to have the optimized version. I'll run a benchmark to see if it makes a difference  since this call is for every doc in the concurrent case